jerry-core/ecma/base/ecma-helpers-external-pointers.c

```diff
@@ -27,61 +27,70 @@
  * @{
  */
 
+/**
+ * Get the number of weak references of the native pointer
+ */
+#define ECMA_NATIVE_POINTER_GET_REFERENCES(info_p) \
+  ((info_p) != NULL ? ((const jerry_object_native_info_t *) (info_p))->number_of_references \
```

Comment:
AFAIK this type here is a layering violation. Why do we use API type here, can't we keep the ecma typedef?

Reply:
I deleted the ecma typedef, this is not a good way to handle these situations. Too easy to make mistakes when you have to maintain two copies of the same thing. The gc already uses `jerryscript.h`, and I think any .c file should be allowed to use it. `jcontext.h` also includes it